### PR TITLE
feat: Add transformer spec reference docs.

### PR DESCRIPTION
**Pull request by marianogappa**

### CODE REVIEW

File: website/pages/docs/reference/transformer-spec.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Name of the plugin. If you have multiple transformer plugins, this must be unique.
+
+The name field may be used to uniquely identify a particular transformer configuration. For example, if you have two configs for the basic plugin for transforming a source table differently in each of two different destination databases, one may be named `basic-1` and the other `basic-2`. In this case, the `path` option below must be used to specify the download path for the plugin.
+
+### registry
+
+(`string`, optional, default: `cloudquery`, available: `github`, `cloudquery`, `local`, `grpc`, `docker`)
+
+- `cloudquery`: CloudQuery will look for and download the plugin from the official CloudQuery registry, and then execute it.
+- `github`: **Deprecated**. CloudQuery will look for and download the plugin from GitHub, and then execute it.
```
Comment (disq) [inline website/pages/docs/reference/transformer-spec.md:44]: Should we include/document deprecated registries while adding a new kind?

Comment (erezrokah) [reply website/pages/docs/reference/transformer-spec.md:44]: Yeah I think we can remove the `github` one it won't work for this plugin anyways

File: website/pages/docs/reference/transformer-spec.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- `local`: CloudQuery will execute the plugin from a local path.
+- `grpc`: mostly useful in debug mode when plugin is already running in a different terminal, CloudQuery will connect to the gRPC plugin server directly without spawning the process.
+- `docker`: CloudQuery will run the plugin in a Docker container. This is most useful for plugins written in Python, as they do not support the `local`, `github` and `cloudquery` registries.
```
Comment (erezrokah) [inline website/pages/docs/reference/transformer-spec.md:47]: Same here `docker` is not supported for this type of plugins yet

File: website/pages/docs/reference/transformer-spec.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+(`string`, required)
+
+`version` must be a valid [SemVer](https://semver.org/), e.g. `vMajor.Minor.Patch`. You can find all official plugin versions under [our GitHub releases page](https://github.com/cloudquery/cloudquery/releases), and for community plugins you can find it in the relevant community repository.
+
+### spec
+
+(`object`, optional)
+
+Plugin specific configurations. Visit [transformer plugins](https://hub.cloudquery.io/plugins/transformer) documentation for more information.
```
[the remark this answers is not in the document]
Comment (marianogappa) [reply website/pages/docs/reference/transformer-spec.md:67]: This is find now right? (incognito screenshot)
Screenshot 2024-08-05 at 14 45 20

File: website/pages/docs/reference/transformer-spec.md (hunk 2 above)
Comment (marianogappa) [reply website/pages/docs/reference/transformer-spec.md]: Removed

File: website/pages/docs/reference/transformer-spec.md (hunk 1 above)
Comment (marianogappa) [reply website/pages/docs/reference/transformer-spec.md]: Removed

**Pull request merged**